docs/DesignDocs/AttestationApiRequirements.md

```diff
@@ -0,0 +1,317 @@
+  necessary endorsements could not be obtained
+* Specified-format-not-supported indicates that the specified
+  Evidence format is not supported
+* Parse-error indicates that the Evidence buffer provided could not be
+  correctly parsed
+* Other failure indicates that the call failed for any other reason
+
+This is the main attestation functionality of the Verifier.
+A pure Verifier would pass the output handle to GetAttestationResults
+in order to generated signed AttestationResults.  A combined
+Verifier/Relying Party could simply use it with GetClaimValue (see below).
+
+## GetAttestationResults call
+
+Inputs:
+
+* Claim set HANDLE
+* Requested Attestation Results format (e.g., CWT, JWT, X.509, etc.).
+  Defaults to letting the plugin choose one.
+
+Outputs:
+
+* Attestation Results buffer OCTET STRING
+* Format used for Attestation Results buffer
+
+Return status codes:
+
+* Success indicates that Attestation Results were generated as requested
+* Invalid-handle indicates that the claim set handle is not valid
+* Requested-format-not-supported indicates that the requested
+  Attestation Results format could not be used
+* Other failure indicates that the call failed for any other reason
+
+# Relying Party APIs
+
+## SetAttestationResultsAppraisalPolicy call
+
+Inputs:
+
+* Attestation Results appraisal policy buffer OCTET STRING
+* Attestation Results appraisal policy buffer format
+
+Outputs:
+
+* Attestation Results appraisal policy HANDLE
+
+Return status codes:
+
+* Success indicates that the appraisal policy was successfully stored
+  and a handle was generated
+* Specified-format-not-supported indicates that the buffer format specified
+  is not supported
+* Parse-error indicates that the buffer provided could not be correctly
+  parsed
+* Other failure indicates that the call failed for any other reason
+
+This call sets an appraisal policy to be used in subsequent
+AppraiseAttestationResults calls.
+
+## AppraiseAttestationResults call
+
+Inputs:
+
+* Attestation Results appraisal policy HANDLE
+* Attestation Results buffer OCTET STRING
+* Format used for Attestation Results buffer
+
+Outputs:
+
+* Claim set HANDLE
+
+Return status codes:
+
+* Success indicates that the Attestation Results were recognized as trusted
+  and a Claim set handle was generated as requested
+* Invalid-handle indicates that the appraisal policy handle is not valid
+* Specified-format-not-supported indicates that the specified
+  Attestation Results format is not supported
+* Parse-error indicates that the Attestation Results buffer could
+  not parse as a legal encoding in the specified Attestation Results format
+* Unauthorized-Results indicates that the Attestation Results did not pass
+  appraisal, such as coming from a Verifier that the Relying Party does not
+  trust, or being past its validity period
+* Other failure indicates that the call failed for any other reason
+
+## GetClaimValue call
```

Comment:
>GetClaimValue call [](start = 3, length = 18)

This call assumes the caller know the claim ID. It will be helpful to have a way to enumerate all claims.

Reply:
Great point, added.